osu.Game.Rulesets.Osu/Difficulty/MathUtil/PoissonBinomial.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using MathNet.Numerics.Distributions;
+
+namespace osu.Game.Rulesets.Osu.Difficulty.MathUtil
+{
+    /// <summary>
+    /// approximate poisson binomial CDF defined by miss probabilities
+    /// see "Refined Normal Approximation (RNA)" from
+    /// https://www.researchgate.net/publication/257017356_On_computing_the_distribution_function_for_the_Poisson_binomial_distribution
```

Comment:
I have reservations to sites like ResearchGate due to the possibility of them gating access to publications at some point in the future. If there was a reference available in a clearly open-access journal like arxiv, then I'd much prefer that.